worker/src/jobs/sync_repo.py

```
@@ -0,0 +1,179 @@
+from .scan_source import apply_deletions
+
+_UNCLASSIFIED = "unclassified/needs-review"
+_ROLE = {"file": "leaf", "module": "group", "repo": "root"}
+_REPO_PATH = "."   # codesum's relpath sentinel for the repo/root artifact
+
+
+def _git_changed_files(root_path: str, base_sha: str) -> tuple[set | None, set]:
+    """(changed, deleted) file relpaths between base_sha and HEAD via git. `changed` is
+    None when the diff can't be computed (→ caller does a full re-summarize)."""
+    try:
+        out = subprocess.run(["git", "-C", root_path, "diff", "--name-status", f"{base_sha}..HEAD"],
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)